### PR TITLE
minor cleanup in graph errs to wrap up

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #4685

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/graph/errors.go [hunk 1]
```diff
@@ -114,47 +114,83 @@ func stackWithCoreErr(ctx context.Context, err error, traceDepth int) error {
 		return nil
 	}
 
+	ode := parseODataErr(err)
+
 	switch {
-	case isErrBadJWTToken(err):
+	case isErrBadJWTToken(ode, err):
```
Comment (meain) [inline src/pkg/services/m365/api/graph/errors.go:120]: I'm missing the context around this change. Why was this necessary?

**Review by ashmrtn — APPROVED** [review]
